llvm/unittests/Target/AIE/AIEScheduleDAGMITest.cpp

```diff
@@ -88,4 +97,47 @@ TEST_F(AIEScheduleDAGMITest, SchedWithDelta) {
   EXPECT_EQ(MISeq(*MBB), MISeq({MI2, MI1, MI0}));
 }
 
+/// Verify the behavior of computeAndFinalizeBundles. In pre-RA scheduling,
+/// it should also support the case where the SU's ReadyCycle is out-of-date
+/// due to it being moved by GenericScheduler::reschedulePhysReg().
+TEST_F(AIEScheduleDAGMITest, SchedThenMove) {
+  auto *MI0 = appendPlainInstr();
+  auto *MI1 = appendPlainInstr();
+  auto *MI2 = appendPlainInstr();
+
+  initializeScheduler(/*IsPreRA=*/true);
+  SchedBoundary &Bot = Scheduler->getSchedZone();
+
+  // Mark all instructions as available.
+  for (auto *MI : {MI0, MI1, MI2})
+    Bot.releaseNode(Scheduler->getSUnit(MI), /*ReadyCycle=*/0, false);
+
+  Scheduler->scheduleInstr(MI2, Bot);
+  Bot.bumpCycle(2);
+  Scheduler->scheduleInstr(MI1, Bot);
+  Bot.bumpCycle(3);
+  Scheduler->scheduleInstr(MI0, Bot);
+
+  EXPECT_EQ(MISeq(*MBB), MISeq({MI0, MI1, MI2}));
+
+  // Make sure the bundles are computed as expected, c1 should be empty.
+  const AIEBaseMCFormats *Fmts = nullptr;
+  std::vector<AIE::MachineBundle> ExpectedBundles = {
+      AIE::MachineBundle({MI0}, Fmts),  // c3
+      AIE::MachineBundle({MI1}, Fmts),  // c2
+      AIE::MachineBundle({}, Fmts),     // c1
+      AIE::MachineBundle({MI2}, Fmts)}; // c0
+  EXPECT_EQ(AIE::computeAndFinalizeBundles(Bot), ExpectedBundles);
+
+  // Now move an instruction to simulate GenericScheduler::reschedulePhysReg.
```

Comment:
perhaps elaborate the action of moveInstruction here, e.g. 'Now move MI2 before MI1'